ubuntu-22.04-arm runners fail consistently

All 3 remaining `ubuntu-22.04-arm` GB runners fail consistently. Updating corresponding actions seems to not fix it. The information on what fails unavailable.

> 2026-05-31T12:21:32.0018833Z Current runner version: '2.334.0'
2026-05-31T12:21:32.0049715Z ##[group]Runner Image Provisioner
2026-05-31T12:21:32.0051025Z Hosted Compute Agent
2026-05-31T12:21:32.0051631Z Version: 20260213.493
2026-05-31T12:21:32.0052369Z Commit: 5c115507f6dd24b8de37d8bbe0bb4509d0cc0fa3
2026-05-31T12:21:32.0053113Z Build Date: 2026-02-13T00:28:41Z
2026-05-31T12:21:32.0053847Z Worker ID: {066a1333-1394-4247-bfd0-dc2618055da7}
2026-05-31T12:21:32.0054620Z Azure Region: eastus2
2026-05-31T12:21:32.0055197Z ##[endgroup]
2026-05-31T12:21:32.0057238Z ##[group]VM Image
2026-05-31T12:21:32.0058063Z - OS: Linux (arm64)
2026-05-31T12:21:32.0058678Z - Source: Partner
2026-05-31T12:21:32.0059584Z - Name: Ubuntu 22.04 by Arm Limited
2026-05-31T12:21:32.0060353Z - Version: 20260318.46.1
2026-05-31T12:21:32.0061356Z - Included Software: https://github.com/actions/partner-runner-images/blob/main/images/arm-ubuntu-22-image.md
2026-05-31T12:21:32.0062471Z ##[endgroup]
2026-05-31T12:21:32.0065407Z ##[group]GITHUB_TOKEN Permissions
2026-05-31T12:21:32.0068559Z Actions: read
2026-05-31T12:21:32.0069712Z ArtifactMetadata: read
2026-05-31T12:21:32.0070427Z Attestations: read
2026-05-31T12:21:32.0070997Z Checks: read
2026-05-31T12:21:32.0071744Z CodeQuality: read
2026-05-31T12:21:32.0072344Z Contents: read
2026-05-31T12:21:32.0072954Z Deployments: read
2026-05-31T12:21:32.0073560Z Discussions: read
2026-05-31T12:21:32.0074130Z Issues: read
2026-05-31T12:21:32.0074754Z Metadata: read
2026-05-31T12:21:32.0075318Z Models: read
2026-05-31T12:21:32.0075824Z Packages: read
2026-05-31T12:21:32.0076492Z Pages: read
2026-05-31T12:21:32.0077021Z PullRequests: read
2026-05-31T12:21:32.0077675Z RepositoryProjects: read
2026-05-31T12:21:32.0078312Z SecurityEvents: read
2026-05-31T12:21:32.0079096Z Statuses: read
2026-05-31T12:21:32.0080032Z VulnerabilityAlerts: read
2026-05-31T12:21:32.0080649Z ##[endgroup]
2026-05-31T12:21:32.0083813Z Secret source: None
2026-05-31T12:21:32.0085000Z Prepare workflow directory
2026-05-31T12:21:32.0440945Z Prepare all required actions
2026-05-31T12:21:32.0479183Z Getting action download info
2026-05-31T12:21:32.3897866Z Download action repository 'actions/checkout@v6' (SHA:de0fac2e4500dabe0009e67214ff5f5447ce83dd)
2026-05-31T12:21:32.8391626Z Complete job name: Build on ubuntu-22.04-arm clang-15
2026-05-31T12:21:32.9280784Z ##[group]Run actions/checkout@v6
2026-05-31T12:21:32.9282565Z with:
2026-05-31T12:21:32.9283300Z   submodules: recursive
2026-05-31T12:21:32.9284088Z   repository: gnuradio/volk
2026-05-31T12:21:32.9289235Z   token: ***
2026-05-31T12:21:32.9290242Z   ssh-strict: true
2026-05-31T12:21:32.9290981Z   ssh-user: git
2026-05-31T12:21:32.9291668Z   persist-credentials: true
2026-05-31T12:21:32.9292413Z   clean: true
2026-05-31T12:21:32.9293114Z   sparse-checkout-cone-mode: true
2026-05-31T12:21:32.9293848Z   fetch-depth: 1
2026-05-31T12:21:32.9294540Z   fetch-tags: false
2026-05-31T12:21:32.9295228Z   show-progress: true
2026-05-31T12:21:32.9295880Z   lfs: false
2026-05-31T12:21:32.9296572Z   set-safe-directory: true
2026-05-31T12:21:32.9297986Z ##[endgroup]
2026-05-31T12:21:33.0142214Z Syncing repository: gnuradio/volk
2026-05-31T12:21:33.0144615Z ##[group]Getting Git version info
2026-05-31T12:21:33.0145748Z Working directory is '/home/runner/work/volk/volk'
2026-05-31T12:21:33.0147458Z [command]/usr/bin/git version
2026-05-31T12:21:33.1074720Z git version 2.53.0
2026-05-31T12:21:33.1096471Z ##[endgroup]
2026-05-31T12:21:33.1111592Z Temporarily overriding HOME='/home/runner/work/_temp/e596756b-aa3c-4568-a475-e098ec9d39af' before making global git config changes
2026-05-31T12:21:33.1113560Z Adding repository directory to the temporary git global config as a safe directory
2026-05-31T12:21:33.1116434Z [command]/usr/bin/git config --global --add safe.directory /home/runner/work/volk/volk
2026-05-31T12:21:33.1239842Z Deleting the contents of '/home/runner/work/volk/volk'
2026-05-31T12:21:33.1244585Z ##[group]Initializing the repository
2026-05-31T12:21:33.1248943Z [command]/usr/bin/git init /home/runner/work/volk/volk
2026-05-31T12:21:33.1761550Z hint: Using 'master' as the name for the initial branch. This default branch name
2026-05-31T12:21:33.1763007Z hint: will change to "main" in Git 3.0. To configure the initial branch name
2026-05-31T12:21:33.1764249Z hint: to use in all of your new repositories, which will suppress this warning,
2026-05-31T12:21:33.1765360Z hint: call:
2026-05-31T12:21:33.1766021Z hint:
2026-05-31T12:21:33.1766839Z hint: 	git config --global init.defaultBranch <name>
2026-05-31T12:21:33.1767664Z hint:
2026-05-31T12:21:33.1768467Z hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
2026-05-31T12:21:33.1770062Z hint: 'development'. The just-created branch can be renamed via this command:
2026-05-31T12:21:33.1771039Z hint:
2026-05-31T12:21:33.1771770Z hint: 	git branch -m <name>
2026-05-31T12:21:33.1772460Z hint:
2026-05-31T12:21:33.1773320Z hint: Disable this message with "git config set advice.defaultBranchName false"
2026-05-31T12:21:33.1774581Z Initialized empty Git repository in /home/runner/work/volk/volk/.git/
2026-05-31T12:21:33.1777109Z [command]/usr/bin/git remote add origin https://github.com/gnuradio/volk
2026-05-31T12:21:33.1868274Z ##[endgroup]
2026-05-31T12:21:33.1869917Z ##[group]Disabling automatic garbage collection
2026-05-31T12:21:33.1871543Z [command]/usr/bin/git config --local gc.auto 0
2026-05-31T12:21:33.1899119Z ##[endgroup]
2026-05-31T12:21:33.1900646Z ##[group]Setting up auth
2026-05-31T12:21:33.1901396Z Removing SSH command configuration
2026-05-31T12:21:33.1905315Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2026-05-31T12:21:33.1932798Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2026-05-31T12:21:33.3538672Z Removing HTTP extra header
2026-05-31T12:21:33.3543236Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2026-05-31T12:21:33.3570839Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2026-05-31T12:21:33.3738085Z Removing includeIf entries pointing to credentials config files
2026-05-31T12:21:33.3742582Z [command]/usr/bin/git config --local --name-only --get-regexp ^includeIf\.gitdir:
2026-05-31T12:21:33.3770353Z [command]/usr/bin/git submodule foreach --recursive git config --local --show-origin --name-only --get-regexp remote.origin.url
2026-05-31T12:21:33.3946805Z [command]/usr/bin/git config --file /home/runner/work/_temp/git-credentials-71a64b66-0d3e-4e6c-b40f-3f7910b997d6.config http.https://github.com/.extraheader AUTHORIZATION: basic ***
2026-05-31T12:21:33.3979090Z [command]/usr/bin/git config --local includeIf.gitdir:/home/runner/work/volk/volk/.git.path /home/runner/work/_temp/git-credentials-71a64b66-0d3e-4e6c-b40f-3f7910b997d6.config
2026-05-31T12:21:33.4006012Z [command]/usr/bin/git config --local includeIf.gitdir:/home/runner/work/volk/volk/.git/worktrees/*.path /home/runner/work/_temp/git-credentials-71a64b66-0d3e-4e6c-b40f-3f7910b997d6.config
2026-05-31T12:21:33.4030439Z [command]/usr/bin/git config --local includeIf.gitdir:/github/workspace/.git.path /github/runner_temp/git-credentials-71a64b66-0d3e-4e6c-b40f-3f7910b997d6.config
2026-05-31T12:21:33.4055162Z [command]/usr/bin/git config --local includeIf.gitdir:/github/workspace/.git/worktrees/*.path /github/runner_temp/git-credentials-71a64b66-0d3e-4e6c-b40f-3f7910b997d6.config
2026-05-31T12:21:33.4076846Z ##[endgroup]
2026-05-31T12:21:33.4078539Z ##[group]Fetching the repository
2026-05-31T12:21:33.4084831Z [command]/usr/bin/git -c protocol.version=2 fetch --no-tags --prune --no-recurse-submodules --depth=1 origin +c5aa7d4beb9037a980f99be9db905ee7c4eba6a5:refs/remotes/pull/864/merge
2026-05-31T12:21:33.8398224Z From https://github.com/gnuradio/volk
2026-05-31T12:21:33.8401162Z  * [new ref]         c5aa7d4beb9037a980f99be9db905ee7c4eba6a5 -> pull/864/merge
2026-05-31T12:21:33.8462582Z ##[endgroup]
2026-05-31T12:21:33.8464314Z ##[group]Determining the checkout info
2026-05-31T12:21:33.8466101Z ##[endgroup]
2026-05-31T12:21:33.8468701Z [command]/usr/bin/git sparse-checkout disable
2026-05-31T12:21:33.8564516Z [command]/usr/bin/git config --local --unset-all extensions.worktreeConfig
2026-05-31T12:21:33.8588001Z ##[group]Checking out the ref
2026-05-31T12:21:33.8590007Z [command]/usr/bin/git checkout --progress --force refs/remotes/pull/864/merge
2026-05-31T12:21:33.8834640Z Note: switching to 'refs/remotes/pull/864/merge'.
2026-05-31T12:21:33.8835643Z 
2026-05-31T12:21:33.8836424Z You are in 'detached HEAD' state. You can look around, make experimental
2026-05-31T12:21:33.8838351Z changes and commit them, and you can discard any commits you make in this
2026-05-31T12:21:33.8840251Z state without impacting any branches by switching back to a branch.
2026-05-31T12:21:33.8841259Z 
2026-05-31T12:21:33.8841989Z If you want to create a new branch to retain commits you create, you may
2026-05-31T12:21:33.8843721Z do so (now or later) by using -c with the switch command. Example:
2026-05-31T12:21:33.8844632Z 
2026-05-31T12:21:33.8845015Z   git switch -c <new-branch-name>
2026-05-31T12:21:33.8845717Z 
2026-05-31T12:21:33.8846176Z Or undo this operation with:
2026-05-31T12:21:33.8846809Z 
2026-05-31T12:21:33.8847173Z   git switch -
2026-05-31T12:21:33.8847607Z 
2026-05-31T12:21:33.8848372Z Turn off this advice by setting config variable advice.detachedHead to false
2026-05-31T12:21:33.8849499Z 
2026-05-31T12:21:33.8850745Z HEAD is now at c5aa7d4 Merge 60cd6544412327de897283fb9c394611ad07dbb6 into e97e1fb4dadd138a5842a8bcffb38e57b067334d
2026-05-31T12:21:33.8854342Z ##[endgroup]
2026-05-31T12:21:33.8855701Z ##[group]Setting up auth for fetching submodules
2026-05-31T12:21:33.8858568Z [command]/usr/bin/git config --file /home/runner/work/_temp/git-credentials-71a64b66-0d3e-4e6c-b40f-3f7910b997d6.config http.https://github.com/.extraheader AUTHORIZATION: basic ***
2026-05-31T12:21:33.8880693Z [command]/usr/bin/git config --global include.path /home/runner/work/_temp/git-credentials-71a64b66-0d3e-4e6c-b40f-3f7910b997d6.config
2026-05-31T12:21:33.8903517Z [command]/usr/bin/git config --global --unset-all url.https://github.com/.insteadOf
2026-05-31T12:21:33.8927499Z [command]/usr/bin/git config --global --add url.https://github.com/.insteadOf git@github.com:
2026-05-31T12:21:33.8956390Z [command]/usr/bin/git config --global --add url.https://github.com/.insteadOf org-1278659@github.com:
2026-05-31T12:21:33.9020843Z ##[endgroup]
2026-05-31T12:21:33.9022252Z ##[group]Fetching submodules
2026-05-31T12:21:33.9023776Z [command]/usr/bin/git submodule sync --recursive
2026-05-31T12:21:33.9245344Z [command]/usr/bin/git -c protocol.version=2 submodule update --init --force --depth=1 --recursive
2026-05-31T12:21:33.9431337Z Submodule 'cpu_features' (https://github.com/google/cpu_features.git) registered for path 'cpu_features'
2026-05-31T12:21:33.9487950Z Cloning into '/home/runner/work/volk/volk/cpu_features'...
2026-05-31T12:21:34.7481842Z From https://github.com/google/cpu_features
2026-05-31T12:21:34.7484299Z  * branch            ba4bffa86cbb5456bdb34426ad22b9551278e2c0 -> FETCH_HEAD
2026-05-31T12:21:34.7560798Z Submodule path 'cpu_features': checked out 'ba4bffa86cbb5456bdb34426ad22b9551278e2c0'
2026-05-31T12:21:34.7585159Z [command]/usr/bin/git submodule foreach --recursive git config --local gc.auto 0
2026-05-31T12:21:34.7751546Z Entering 'cpu_features'
2026-05-31T12:21:34.7787794Z ##[endgroup]
2026-05-31T12:21:34.7788325Z ##[group]Persisting credentials for submodules
2026-05-31T12:21:34.7794658Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'url\.https\:\/\/github\.com\/\.insteadOf' && git config --local --unset-all 'url.https://github.com/.insteadOf' || :"
2026-05-31T12:21:34.7962236Z Entering 'cpu_features'
2026-05-31T12:21:34.8005959Z [command]/usr/bin/git submodule foreach --recursive git config --local --show-origin --name-only --get-regexp remote.origin.url
2026-05-31T12:21:34.8170381Z Entering 'cpu_features'
2026-05-31T12:21:34.8184358Z file:/home/runner/work/volk/volk/.git/modules/cpu_features/config	remote.origin.url
2026-05-31T12:21:34.8209838Z [command]/usr/bin/git config --file /home/runner/work/volk/volk/.git/modules/cpu_features/config includeIf.gitdir:/home/runner/work/volk/volk/.git/modules/cpu_features.path /home/runner/work/_temp/git-credentials-71a64b66-0d3e-4e6c-b40f-3f7910b997d6.config
2026-05-31T12:21:34.8236301Z [command]/usr/bin/git config --file /home/runner/work/volk/volk/.git/modules/cpu_features/config includeIf.gitdir:/github/workspace/.git/modules/cpu_features.path /github/runner_temp/git-credentials-71a64b66-0d3e-4e6c-b40f-3f7910b997d6.config
2026-05-31T12:21:34.8262607Z [command]/usr/bin/git submodule foreach --recursive git config --local --add 'url.https://github.com/.insteadOf' 'git@github.com:'
2026-05-31T12:21:34.8428602Z Entering 'cpu_features'
2026-05-31T12:21:34.8469186Z [command]/usr/bin/git submodule foreach --recursive git config --local --add 'url.https://github.com/.insteadOf' 'org-1278659@github.com:'
2026-05-31T12:21:34.8639267Z Entering 'cpu_features'
2026-05-31T12:21:34.8677315Z ##[endgroup]
2026-05-31T12:21:34.8710396Z [command]/usr/bin/git log -1 --format=%H
2026-05-31T12:21:34.8730936Z c5aa7d4beb9037a980f99be9db905ee7c4eba6a5
2026-05-31T12:21:34.8950038Z ##[group]Run sudo apt install python3-mako liborc-dev libgtest-dev libfmt-dev clang-15
2026-05-31T12:21:34.8950882Z [36;1msudo apt install python3-mako liborc-dev libgtest-dev libfmt-dev clang-15[0m
2026-05-31T12:21:34.8982275Z shell: /usr/bin/bash -e {0}
2026-05-31T12:21:34.8982684Z ##[endgroup]
2026-05-31T12:21:34.9289897Z 
2026-05-31T12:21:34.9290549Z WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
2026-05-31T12:21:34.9290983Z 
2026-05-31T12:21:40.5982072Z Reading package lists...
2026-05-31T12:21:41.0715456Z Building dependency tree...
2026-05-31T12:21:41.0777030Z Reading state information...
2026-05-31T12:21:41.2917843Z clang-15 is already the newest version (1:15.0.7-0ubuntu0.22.04.3).
2026-05-31T12:21:41.2918772Z The following additional packages will be installed:
2026-05-31T12:21:41.2922822Z   googletest libfmt8 liborc-0.4-0 liborc-0.4-dev-bin
2026-05-31T12:21:41.2930515Z Suggested packages:
2026-05-31T12:21:41.2931125Z   libfmt-doc liborc-0.4-doc python3-beaker python-mako-doc
2026-05-31T12:21:41.3700156Z The following NEW packages will be installed:
2026-05-31T12:21:41.3700785Z   googletest libfmt-dev libfmt8 libgtest-dev liborc-0.4-0 liborc-0.4-dev
2026-05-31T12:21:41.3706178Z   liborc-0.4-dev-bin python3-mako
2026-05-31T12:21:41.4153353Z 0 upgraded, 8 newly installed, 0 to remove and 14 not upgraded.
2026-05-31T12:21:41.6635630Z Need to get 1695 kB of archives.
2026-05-31T12:21:41.6636166Z After this operation, 9817 kB of additional disk space will be used.
2026-05-31T12:21:41.6636875Z Get:1 http://ports.ubuntu.com/ubuntu-ports jammy/universe arm64 googletest all 1.11.0-3 [541 kB]
2026-05-31T12:21:41.6935274Z Get:2 http://ports.ubuntu.com/ubuntu-ports jammy/universe arm64 libfmt8 arm64 8.1.1+ds1-2 [56.9 kB]
2026-05-31T12:21:41.7050207Z Get:3 http://ports.ubuntu.com/ubuntu-ports jammy/universe arm64 libgtest-dev arm64 1.11.0-3 [248 kB]
2026-05-31T12:21:41.7180268Z Get:4 http://ports.ubuntu.com/ubuntu-ports jammy-updates/main arm64 liborc-0.4-0 arm64 1:0.4.32-2ubuntu0.1 [222 kB]
2026-05-31T12:21:41.7316134Z Get:5 http://ports.ubuntu.com/ubuntu-ports jammy-updates/main arm64 liborc-0.4-dev-bin arm64 1:0.4.32-2ubuntu0.1 [229 kB]
2026-05-31T12:21:41.7442916Z Get:6 http://ports.ubuntu.com/ubuntu-ports jammy-updates/main arm64 liborc-0.4-dev arm64 1:0.4.32-2ubuntu0.1 [218 kB]
2026-05-31T12:21:41.8709469Z Ign:7 http://ports.ubuntu.com/ubuntu-ports jammy-updates/main arm64 python3-mako all 1.1.3+ds1-2ubuntu0.1
2026-05-31T12:21:41.8822444Z Get:8 http://ports.ubuntu.com/ubuntu-ports jammy/universe arm64 libfmt-dev arm64 8.1.1+ds1-2 [118 kB]
2026-05-31T12:21:41.8947373Z Err:7 http://ports.ubuntu.com/ubuntu-ports jammy-updates/main arm64 python3-mako all 1.1.3+ds1-2ubuntu0.1
2026-05-31T12:21:41.8947945Z   404  Not Found [IP: 104.20.28.246 80]
2026-05-31T12:21:41.8957210Z Fetched 1634 kB in 0s (3697 kB/s)
2026-05-31T12:21:41.9370584Z E: Failed to fetch http://ports.ubuntu.com/ubuntu-ports/pool/main/m/mako/python3-mako_1.1.3%2bds1-2ubuntu0.1_all.deb  404  Not Found [IP: 104.20.28.246 80]
2026-05-31T12:21:41.9371522Z E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
2026-05-31T12:21:41.9409994Z ##[error]Process completed with exit code 100.
2026-05-31T12:21:41.9566981Z Post job cleanup.
2026-05-31T12:21:42.0251828Z [command]/usr/bin/git version
2026-05-31T12:21:42.0283097Z git version 2.53.0
2026-05-31T12:21:42.0318202Z Temporarily overriding HOME='/home/runner/work/_temp/842f5bb9-49ef-4753-bfca-082fe0015cbb' before making global git config changes
2026-05-31T12:21:42.0319600Z Adding repository directory to the temporary git global config as a safe directory
2026-05-31T12:21:42.0322788Z [command]/usr/bin/git config --global --add safe.directory /home/runner/work/volk/volk
2026-05-31T12:21:42.0348545Z Removing SSH command configuration
2026-05-31T12:21:42.0354611Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2026-05-31T12:21:42.0383065Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2026-05-31T12:21:42.0545877Z Entering 'cpu_features'
2026-05-31T12:21:42.0588982Z Removing HTTP extra header
2026-05-31T12:21:42.0593435Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2026-05-31T12:21:42.0622964Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2026-05-31T12:21:42.0793319Z Entering 'cpu_features'
2026-05-31T12:21:42.0835653Z Removing includeIf entries pointing to credentials config files
2026-05-31T12:21:42.0842390Z [command]/usr/bin/git config --local --name-only --get-regexp ^includeIf\.gitdir:
2026-05-31T12:21:42.0862057Z includeif.gitdir:/home/runner/work/volk/volk/.git.path
2026-05-31T12:21:42.0863105Z includeif.gitdir:/home/runner/work/volk/volk/.git/worktrees/*.path
2026-05-31T12:21:42.0863630Z includeif.gitdir:/github/workspace/.git.path
2026-05-31T12:21:42.0864156Z includeif.gitdir:/github/workspace/.git/worktrees/*.path
2026-05-31T12:21:42.0870332Z [command]/usr/bin/git config --local --get-all includeif.gitdir:/home/runner/work/volk/volk/.git.path
2026-05-31T12:21:42.0889370Z /home/runner/work/_temp/git-credentials-71a64b66-0d3e-4e6c-b40f-3f7910b997d6.config
2026-05-31T12:21:42.0900731Z [command]/usr/bin/git config --local --unset includeif.gitdir:/home/runner/work/volk/volk/.git.path /home/runner/work/_temp/git-credentials-71a64b66-0d3e-4e6c-b40f-3f7910b997d6.config
2026-05-31T12:21:42.0929199Z [command]/usr/bin/git config --local --get-all includeif.gitdir:/home/runner/work/volk/volk/.git/worktrees/*.path
2026-05-31T12:21:42.0948382Z /home/runner/work/_temp/git-credentials-71a64b66-0d3e-4e6c-b40f-3f7910b997d6.config
2026-05-31T12:21:42.0957181Z [command]/usr/bin/git config --local --unset includeif.gitdir:/home/runner/work/volk/volk/.git/worktrees/*.path /home/runner/work/_temp/git-credentials-71a64b66-0d3e-4e6c-b40f-3f7910b997d6.config
2026-05-31T12:21:42.0984823Z [command]/usr/bin/git config --local --get-all includeif.gitdir:/github/workspace/.git.path
2026-05-31T12:21:42.1002633Z /github/runner_temp/git-credentials-71a64b66-0d3e-4e6c-b40f-3f7910b997d6.config
2026-05-31T12:21:42.1009574Z [command]/usr/bin/git config --local --unset includeif.gitdir:/github/workspace/.git.path /github/runner_temp/git-credentials-71a64b66-0d3e-4e6c-b40f-3f7910b997d6.config
2026-05-31T12:21:42.1035503Z [command]/usr/bin/git config --local --get-all includeif.gitdir:/github/workspace/.git/worktrees/*.path
2026-05-31T12:21:42.1053184Z /github/runner_temp/git-credentials-71a64b66-0d3e-4e6c-b40f-3f7910b997d6.config
2026-05-31T12:21:42.1060619Z [command]/usr/bin/git config --local --unset includeif.gitdir:/github/workspace/.git/worktrees/*.path /github/runner_temp/git-credentials-71a64b66-0d3e-4e6c-b40f-3f7910b997d6.config
2026-05-31T12:21:42.1087569Z [command]/usr/bin/git submodule foreach --recursive git config --local --show-origin --name-only --get-regexp remote.origin.url
2026-05-31T12:21:42.1255709Z Entering 'cpu_features'
2026-05-31T12:21:42.1268969Z file:/home/runner/work/volk/volk/.git/modules/cpu_features/config	remote.origin.url
2026-05-31T12:21:42.1294531Z [command]/usr/bin/git config --file /home/runner/work/volk/volk/.git/modules/cpu_features/config --name-only --get-regexp ^includeIf\.gitdir:
2026-05-31T12:21:42.1313919Z includeif.gitdir:/home/runner/work/volk/volk/.git/modules/cpu_features.path
2026-05-31T12:21:42.1314618Z includeif.gitdir:/github/workspace/.git/modules/cpu_features.path
2026-05-31T12:21:42.1320830Z [command]/usr/bin/git config --file /home/runner/work/volk/volk/.git/modules/cpu_features/config --get-all includeif.gitdir:/home/runner/work/volk/volk/.git/modules/cpu_features.path
2026-05-31T12:21:42.1338959Z /home/runner/work/_temp/git-credentials-71a64b66-0d3e-4e6c-b40f-3f7910b997d6.config
2026-05-31T12:21:42.1346531Z [command]/usr/bin/git config --file /home/runner/work/volk/volk/.git/modules/cpu_features/config --unset includeif.gitdir:/home/runner/work/volk/volk/.git/modules/cpu_features.path /home/runner/work/_temp/git-credentials-71a64b66-0d3e-4e6c-b40f-3f7910b997d6.config
2026-05-31T12:21:42.1382062Z [command]/usr/bin/git config --file /home/runner/work/volk/volk/.git/modules/cpu_features/config --get-all includeif.gitdir:/github/workspace/.git/modules/cpu_features.path
2026-05-31T12:21:42.1399897Z /github/runner_temp/git-credentials-71a64b66-0d3e-4e6c-b40f-3f7910b997d6.config
2026-05-31T12:21:42.1407262Z [command]/usr/bin/git config --file /home/runner/work/volk/volk/.git/modules/cpu_features/config --unset includeif.gitdir:/github/workspace/.git/modules/cpu_features.path /github/runner_temp/git-credentials-71a64b66-0d3e-4e6c-b40f-3f7910b997d6.config
2026-05-31T12:21:42.1431790Z Removing credentials config '/home/runner/work/_temp/git-credentials-71a64b66-0d3e-4e6c-b40f-3f7910b997d6.config'
2026-05-31T12:21:42.1561909Z Cleaning up orphan processes

Maybe the issue is related to this error:

> 2026-05-31T12:21:41.8947373Z Err:7 http://ports.ubuntu.com/ubuntu-ports jammy-updates/main arm64 python3-mako all 1.1.3+ds1-2ubuntu0.1
2026-05-31T12:21:41.8947945Z 404 Not Found [IP: 104.20.28.246 80]
2026-05-31T12:21:41.8957210Z Fetched 1634 kB in 0s (3697 kB/s)
2026-05-31T12:21:41.9370584Z E: Failed to fetch http://ports.ubuntu.com/ubuntu-ports/pool/main/m/mako/python3-mako_1.1.3%2bds1-2ubuntu0.1_all.deb 404 Not Found [IP: 104.20.28.246 80]
2026-05-31T12:21:41.9371522Z E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
2026-05-31T12:21:41.9409994Z ##[error]Process completed with exit code 100.

This is still buried somewhere in the runner image. There are reports that the arm runners were previously built by ARM (according to the log we use such a version) and they transition to GH owned runner images. We might be stuck in the transition.

--- a/.github/workflows/android_build.yml
+++ b/.github/workflows/android_build.yml
@@ -17,7 +17,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        arch: 
+        arch:
           - { name: armeabi-v7a, allow_fail: false }
           - { name: arm64-v8a, allow_fail: false }
           - { name: x86, allow_fail: false }
@@ -26,38 +26,38 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v4
-      with:
-        submodules: 'recursive'
+      - uses: actions/checkout@v6
+        with:
+          submodules: "recursive"
 
-    - name: Update repositories
-      run: sudo apt update 
-    
-    # All dependencies
-    - name: Install dependencies
-      run: sudo apt install -y cmake python3-mako
+      - name: Update repositories
+        run: sudo apt update
 
-    # Setup Java
-    - uses: actions/setup-java@v4
-      with:
-        distribution: 'temurin'
-        java-version: '17'
+      # All dependencies
+      - name: Install dependencies
+        run: sudo apt install -y cmake python3-mako
 
-    # Setup Android SDK, and auto-accept licenses
-    - name: Install Android SDK
-      run: wget --quiet --output-document=android-sdk.zip https://dl.google.com/android/repository/commandlinetools-linux-11076708_latest.zip && mkdir android-sdk-linux && unzip -qq android-sdk.zip -d android-sdk-linux && export ANDROID_HOME=./android-sdk-linux && echo y | $ANDROID_HOME/cmdline-tools/bin/sdkmanager --sdk_root=android-sdk-linux --update && (echo y; echo y; echo y; echo y; echo y; echo y; echo y; echo y) | $ANDROID_HOME/cmdline-tools/bin/sdkmanager --sdk_root=android-sdk-linux --licenses
+      # Setup Java
+      - uses: actions/setup-java@v5
+        with:
+          distribution: "temurin"
+          java-version: "17"
 
-    # Call SDKManager to install the Android NDK
-    - name: Install Android NDK
-      run: $GITHUB_WORKSPACE/android-sdk-linux/cmdline-tools/bin/sdkmanager --sdk_root=$GITHUB_WORKSPACE/android-sdk-linux --install "ndk;27.2.12479018" --channel=3
+      # Setup Android SDK, and auto-accept licenses
+      - name: Install Android SDK
+        run: wget --quiet --output-document=android-sdk.zip https://dl.google.com/android/repository/commandlinetools-linux-11076708_latest.zip && mkdir android-sdk-linux && unzip -qq android-sdk.zip -d android-sdk-linux && export ANDROID_HOME=./android-sdk-linux && echo y | $ANDROID_HOME/cmdline-tools/bin/sdkmanager --sdk_root=android-sdk-linux --update && (echo y; echo y; echo y; echo y; echo y; echo y; echo y; echo y) | $ANDROID_HOME/cmdline-tools/bin/sdkmanager --sdk_root=android-sdk-linux --licenses
 
-    # Setup build directory
-    - name: Setup ${{ matrix.arch.name }}
-      shell: bash
-      run: cd $GITHUB_WORKSPACE/ && mkdir build && cd build && cmake -DCMAKE_TOOLCHAIN_FILE=$GITHUB_WORKSPACE/android-sdk-linux/ndk/27.2.12479018/build/cmake/android.toolchain.cmake -DANDROID_ABI=${{ matrix.arch.name }} -DANDROID_PLATFORM=android-34 ..
+      # Call SDKManager to install the Android NDK
+      - name: Install Android NDK
+        run: $GITHUB_WORKSPACE/android-sdk-linux/cmdline-tools/bin/sdkmanager --sdk_root=$GITHUB_WORKSPACE/android-sdk-linux --install "ndk;27.2.12479018" --channel=3
 
-    # Build
-    - name: Build ${{ matrix.arch.name }}
-      shell: bash
-      run: cd $GITHUB_WORKSPACE/build && make
-      continue-on-error: ${{ matrix.arch.allow_fail }}
+      # Setup build directory
+      - name: Setup ${{ matrix.arch.name }}
+        shell: bash
+        run: cd $GITHUB_WORKSPACE/ && mkdir build && cd build && cmake -DCMAKE_TOOLCHAIN_FILE=$GITHUB_WORKSPACE/android-sdk-linux/ndk/27.2.12479018/build/cmake/android.toolchain.cmake -DANDROID_ABI=${{ matrix.arch.name }} -DANDROID_PLATFORM=android-34 ..
+
+      # Build
+      - name: Build ${{ matrix.arch.name }}
+        shell: bash
+        run: cd $GITHUB_WORKSPACE/build && make
+        continue-on-error: ${{ matrix.arch.allow_fail }}

--- a/.github/workflows/check-pr-formatting.yml
+++ b/.github/workflows/check-pr-formatting.yml
@@ -11,22 +11,21 @@ name: Check PR Formatting
 on:
   push:
     paths-ignore:
-      - 'tmpl/'
-      - 'include/volk/sse2neon.h'
+      - "tmpl/"
+      - "include/volk/sse2neon.h"
   pull_request:
     paths-ignore:
-      - 'tmpl/'
-      - 'include/volk/sse2neon.h'
+      - "tmpl/"
+      - "include/volk/sse2neon.h"
 
 jobs:
   build:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v4
-    - uses: gnuradio/clang-format-lint-action@v0.5-4
-      with:
-        source: '.'
-        exclude: './tmpl,./include/volk/sse2neon.h'
-        extensions: 'c,cc,cpp,cxx,h,hh'
-
+      - uses: actions/checkout@v6
+      - uses: gnuradio/clang-format-lint-action@v0.5-4
+        with:
+          source: "."
+          exclude: "./tmpl,./include/volk/sse2neon.h"
+          extensions: "c,cc,cpp,cxx,h,hh"

--- a/.github/workflows/publish_docs.yml
+++ b/.github/workflows/publish_docs.yml
@@ -7,7 +7,7 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@v6
         with:
           submodules: "recursive"
       - name: Install dependencies
@@ -29,7 +29,7 @@ jobs:
           SSH_AUTH_SOCK: /tmp/ssh_agent.sock
           TARGET_DIR: "${{ github.ref_type }}/${{ github.ref_name }}"
         run: 'tar -cz build/html/ | ssh ${{ secrets.SSH_USER }}@${{ secrets.SSH_SERVER }} "mkdir -p /www/${{ env.TARGET_DIR }}/$(date +%Y.%m.%d); cd /www/${{ env.TARGET_DIR }}/$(date +%Y.%m.%d); tar --strip-components=2 -xzf -; rm -f /www/${{ env.TARGET_DIR }}/live; cd /www/${{ env.TARGET_DIR }}; ln -sf $(date +%Y.%m.%d) live;"'
-      - uses: actions/upload-artifact@v4
+      - uses: actions/upload-artifact@v7
         with:
           name: volk_docs
           path: build/html/

--- a/.github/workflows/run-tests-rvv.yml
+++ b/.github/workflows/run-tests-rvv.yml
@@ -14,42 +14,40 @@ jobs:
   Tests:
     runs-on: ubuntu-24.04
     steps:
-    - uses: actions/checkout@v4
-      with:
-        submodules: "recursive"
-    - name: Install packages
-      run: |
-        sudo apt-get update -q -y
-        sudo apt-get install -y python3-mako cmake qemu-user-static g++-14-riscv64-linux-gnu clang-18
-        mkdir build
-        cd build
-    - name: Test gcc-14 VLEN=128
-      run: |
-        cd build; rm -rf *
-        CXX=riscv64-linux-gnu-g++-14 CC=riscv64-linux-gnu-gcc-14 VLEN=128 \
-        cmake -DCMAKE_TOOLCHAIN_FILE=../cmake/Toolchains/rv64gcv-linux-gnu.cmake ..
-        make -j$(nproc)
-        ARGS=-V make test
-    - name: Test gcc-14 VLEN=256
-      run: |
-        cd build; rm -rf *
-        CXX=riscv64-linux-gnu-g++-14 CC=riscv64-linux-gnu-gcc-14 VLEN=256 \
-        cmake -DCMAKE_TOOLCHAIN_FILE=../cmake/Toolchains/rv64gcv-linux-gnu.cmake .. -DCMAKE_BUILD_TYPE=Release
-        make -j$(nproc)
-        ARGS=-V make test
-    - name: Test clang-18 VLEN=512
-      run: |
-        cd build; rm -rf *
-        CXX=clang++-18 CC=clang-18 CFLAGS=--target=riscv64-linux-gnu VLEN=512 \
-        cmake -DCMAKE_TOOLCHAIN_FILE=../cmake/Toolchains/rv64gcv-linux-gnu.cmake ..
-        make -j$(nproc)
-        ARGS=-V make test
-    - name: Test clang-18 VLEN=1024
-      run: |
-        cd build; rm -rf *
-        CXX=clang++-18 CC=clang-18 CFLAGS=--target=riscv64-linux-gnu VLEN=1024 \
-        cmake -DCMAKE_TOOLCHAIN_FILE=../cmake/Toolchains/rv64gcv-linux-gnu.cmake .. -DCMAKE_BUILD_TYPE=Release
-        make -j$(nproc)
-        ARGS=-V make test
-
-
+      - uses: actions/checkout@v6
+        with:
+          submodules: "recursive"
+      - name: Install packages
+        run: |
+          sudo apt-get update -q -y
+          sudo apt-get install -y python3-mako cmake qemu-user-static g++-14-riscv64-linux-gnu clang-18
+          mkdir build
+          cd build
+      - name: Test gcc-14 VLEN=128
+        run: |
+          cd build; rm -rf *
+          CXX=riscv64-linux-gnu-g++-14 CC=riscv64-linux-gnu-gcc-14 VLEN=128 \
+          cmake -DCMAKE_TOOLCHAIN_FILE=../cmake/Toolchains/rv64gcv-linux-gnu.cmake ..
+          make -j$(nproc)
+          ARGS=-V make test
+      - name: Test gcc-14 VLEN=256
+        run: |
+          cd build; rm -rf *
+          CXX=riscv64-linux-gnu-g++-14 CC=riscv64-linux-gnu-gcc-14 VLEN=256 \
+          cmake -DCMAKE_TOOLCHAIN_FILE=../cmake/Toolchains/rv64gcv-linux-gnu.cmake .. -DCMAKE_BUILD_TYPE=Release
+          make -j$(nproc)
+          ARGS=-V make test
+      - name: Test clang-18 VLEN=512
+        run: |
+          cd build; rm -rf *
+          CXX=clang++-18 CC=clang-18 CFLAGS=--target=riscv64-linux-gnu VLEN=512 \
+          cmake -DCMAKE_TOOLCHAIN_FILE=../cmake/Toolchains/rv64gcv-linux-gnu.cmake ..
+          make -j$(nproc)
+          ARGS=-V make test
+      - name: Test clang-18 VLEN=1024
+        run: |
+          cd build; rm -rf *
+          CXX=clang++-18 CC=clang-18 CFLAGS=--target=riscv64-linux-gnu VLEN=1024 \
+          cmake -DCMAKE_TOOLCHAIN_FILE=../cmake/Toolchains/rv64gcv-linux-gnu.cmake .. -DCMAKE_BUILD_TYPE=Release
+          make -j$(nproc)
+          ARGS=-V make test

--- a/.github/workflows/run-tests.yml
+++ b/.github/workflows/run-tests.yml
@@ -19,9 +19,9 @@ jobs:
       matrix:
         compiler:
           - { name: g++-11, cc: gcc-11, cxx: g++-11, distro: ubuntu-22.04 }
-          - { name: g++-11, cc: gcc-11, cxx: g++-11, distro: ubuntu-22.04-arm }
+          # - { name: g++-11, cc: gcc-11, cxx: g++-11, distro: ubuntu-22.04-arm }
           - { name: g++-12, cc: gcc-12, cxx: g++-12, distro: ubuntu-22.04 }
-          - { name: g++-12, cc: gcc-12, cxx: g++-12, distro: ubuntu-22.04-arm }
+          # - { name: g++-12, cc: gcc-12, cxx: g++-12, distro: ubuntu-22.04-arm }
           - { name: g++-13, cc: gcc-13, cxx: g++-13, distro: ubuntu-24.04 }
           - { name: g++-13, cc: gcc-13, cxx: g++-13, distro: ubuntu-24.04-arm }
           - { name: g++-14, cc: gcc-14, cxx: g++-14, distro: ubuntu-24.04 }
@@ -29,7 +29,7 @@ jobs:
           - { name: clang-14, cc: clang-14, cxx: clang++-14, distro: ubuntu-22.04 }
           # - { name: clang-14, cc: clang-14, cxx: clang++-14, distro: ubuntu-22.04-arm } # possibly broken runner: https://github.com/actions/runner-images/issues/8659
           - { name: clang-15, cc: clang-15, cxx: clang++-15, distro: ubuntu-22.04 }
-          - { name: clang-15, cc: clang-15, cxx: clang++-15, distro: ubuntu-22.04-arm }
+          # - { name: clang-15, cc: clang-15, cxx: clang++-15, distro: ubuntu-22.04-arm }
           - { name: clang-16, cc: clang-16, cxx: clang++-16, distro: ubuntu-24.04 }
           - { name: clang-16, cc: clang-16, cxx: clang++-16, distro: ubuntu-24.04-arm }
           - { name: clang-17, cc: clang-17, cxx: clang++-17, distro: ubuntu-24.04 }

--- a/.github/workflows/run-tests.yml
+++ b/.github/workflows/run-tests.yml
@@ -42,7 +42,7 @@ jobs:
     runs-on: ${{ matrix.compiler.distro }}
 
     steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@v6
         with:
           submodules: "recursive"
       - name: Install dependencies
@@ -97,7 +97,7 @@ jobs:
             compiler: { name: g++-12, cc: gcc-12, cxx: g++-12 }
 
     steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@v6
         with:
           submodules: "recursive"
       - uses: uraimo/run-on-arch-action@v3
@@ -155,7 +155,7 @@ jobs:
     runs-on: ubuntu-22.04
 
     steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@v6
         with:
           submodules: "recursive"
       - name: dependencies
@@ -179,10 +179,10 @@ jobs:
     runs-on: windows-latest
 
     steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@v6
         with:
           submodules: "recursive"
-      - uses: actions/setup-python@v5
+      - uses: actions/setup-python@v6
         with:
           python-version: "3.12"
       - name: dependencies
@@ -212,7 +212,7 @@ jobs:
   #           python-six
   #           mingw-w64-x86_64-gcc
   #           mingw-w64-x86_64-cmake
-  #     - uses: actions/checkout@v4
+  #     - uses: actions/checkout@v6
   #     - name: Checkout submodules
   #       run: git submodule update --init --recursive
   #     - name: Configure
@@ -235,7 +235,7 @@ jobs:
     runs-on: ${{ matrix.os }}
 
     steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@v6
         with:
           submodules: "recursive"
       - name: dependencies

